607080  - [6.0] snmpd fails to start up due to segfault

Source: upstream, SVN rev. 17852


===================================================================
--- a/agent/mibgroup/disman/schedule/schedCore.c	(revision 17851)
+++ b/agent/mibgroup/disman/schedule/schedCore.c	(revision 17852)
@@ -449,13 +449,15 @@
      */
     if (schedOwner) {
         memcpy(entry->schedOwner, schedOwner, strlen(schedOwner));
+        netsnmp_tdata_row_add_index(row, ASN_OCTET_STR,
+                           entry->schedOwner, strlen(schedOwner));
     }
+    else
+        netsnmp_tdata_row_add_index(row, ASN_OCTET_STR, "", 0 );
+
     memcpy(    entry->schedName,  schedName,  strlen(schedName));
     netsnmp_tdata_row_add_index(row, ASN_OCTET_STR,
-                                (entry->schedOwner || ""),
-                                ((schedOwner) ? strlen(schedOwner) : 0));
-    netsnmp_tdata_row_add_index(row, ASN_OCTET_STR,
-                                entry->schedName,  strlen(schedName));
+                           entry->schedName,  strlen(schedName));
     /*
      * Set the (non-zero) default values in the row data structure.
      */
